JUnit5-README.md

```diff
@@ -0,0 +1,193 @@
+
+You can download JSON definition of already existing resources or downloading JSON for automation in create form.
+
+Azure template allows you to deploy and configure Azure specific resources using native technologies.
+
+#### AWS CloudFormation template
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+public class AWSTest {}
+```
+
+In this case, template is located in resources. You can also provide a raw content or URL and the template shall be downloaded. Resources are deployed to a stack of random name in region defined in `sunstone.properties`
+
+You can use Former2 tool to obtain YAML definition of already existing resources.
+
+CloudFormation template allows you to deploy and configure AWS specific resources using native technologies.
+
+#### JClouds
+TODO
+
+```java
+@Sunstone
+@WithJClouds("test.properties")
+public class JCloudTest {}
+```
+
+In this case, property file is located in resources. You can also provide a raw content or URL and the file shall be downloaded.
+
+Contract for different providers can be found at:
+- [azure-arm-README](azure-arm-README.md)
+- [baremetal-README](baremetal-README.md)
+- [ec2-README](ec2-README.md)
+- [docker-README](docker-README.md)
+- [openstack-README](openstack-README.md)
+- [wildfly-README](wildfly-README.md)
+
+JCloud enables you to share a test among different providers.
+
+### Provide additional configuration
+Sunstone allows you to define a setup task with enabled DI. Add the `@Setup` annotation on test class
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {}
+```
+```java
+
+public class Configure2LCApp extends AbstractSetupTask {
+    
+  @SunstoneResource(of = "eapqe-instance", hint = AWS_EC2_INSTANCE)
+  OnlineManagementClient client;
+
+  @Override
+  public void setup() throws Exception {
+    // configure eap
+  }
+
+  @Override
+  public void cleanup() throws Exception {
+    // clean the configuration
+  }
+}
+```
+
+
+### Use DI for test development
+Sunstone allows you to inject several resources: EAP management client, cloud clients, ... Fields may be either static (injected once) or non-static (injected per test case).
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {
+  @SunstoneResource(of = "eapqe-mdb", hint = AWS_EC2_INSTANCE)
+  static Hostname mdbInstanceHostname;
+}
+```
+
+### Develop tests
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {
+  @SunstoneResource(of = "eapqe-mdb", hint = AWS_EC2_INSTANCE)
+  static Hostname mdbInstanceHostname;
+  
+  @Test
+  public void test() {
+      assert Rest.get(mdbInstanceHostname.get()).code() == 200;
+  }
+}
+```
+
+Users usually have a configuration file `arquillian.xml` on the classpath and its content looks similar to this one:
```

Comment:
I'm not sure what is the correlation here between this new way of deploying EAP/WF and Arquillian. I thought that Arquillian in those types of tests cannot be used.

Reply:
Yeah, forgot about those lines:)